internal/server/server.go

```diff
@@ -19,26 +20,49 @@ var (
 
 // Server contains all the necessary information to run Bifrost
 type Server struct {
-	srv *http.Server
+	certFile string
+	keyFile  string
+	srv      *http.Server
 }
 
 // New creates a new Bifrost server
-func New(port string) *Server {
-	s := &Server{
-		srv: &http.Server{
-			Addr:         net.JoinHostPort("0.0.0.0", port),
-			ReadTimeout:  60 * time.Second,
-			WriteTimeout: 60 * time.Second,
-			IdleTimeout:  30 * time.Second,
+func New(cfg *Config) *Server {
+	server := &http.Server{
+		Addr:         net.JoinHostPort("0.0.0.0", cfg.ServiceSettings.Port),
+		ReadTimeout:  60 * time.Second,
+		WriteTimeout: 60 * time.Second,
+		IdleTimeout:  30 * time.Second,
+	}
+
+	server.TLSConfig = &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
+			tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
 		},
 	}
 
+	s := &Server{
+		certFile: cfg.ServiceSettings.TLSCertFile,
+		keyFile:  cfg.ServiceSettings.TLSKeyFile,
```

Comment:
I'd put the config itself in the `Server` and use the cert and key files from there. I also need the config for later stuff.